docs/develop/typescript-restrictions.md

```diff
@@ -101,6 +101,25 @@ interface MyTable {
 }
 ```
 
+## Constructors don't support Office Scripts APIs and `console` statements
+
+`console` statements and many Office Scripts APIs require synchronization with the Excel workbook. These synchronizations use `await` statements in compiled runtime version of the script. `await` is not supported in constructors. If you need classes with constructors, avoid using Office Scripts APIs or `console` statements in those code blocks.
+
+The following code sample demonstrates this scenario. It generates an error that says "failed to load \[code\] \[library\]".
```

Comment:
Should the documentation point out a work around, such as a static create method?

Reply:
Maybe? This all seems very advanced for our audience.

Reply:
@wandyezj, judging by your [Stack answer](https://stackoverflow.com/a/69846140/9227753), it seems like the guidance should really be "avoid doing Office stuff in constructors". I think we're good just documenting the limitation, particularly the console bits.

Reply:
Is the error printed in the console? Something feels a little off to me about the formatting of the error message here. In the Explicit `any` section, the error is wrapped in code ticks.

Reply:
I switched it to match, but I think we should ultimately decide this based on error message localization. 
@wandyezj, do Office Scripts error messages get localized? If so, we should document them in quotation marks so they get translated (and hopefully match the translated version in the code). Otherwise, code ticks will ensure the strings don't get translated.